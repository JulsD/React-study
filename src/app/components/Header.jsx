import React from 'react';
import Nav from './Nav';
import SearchBox from './SearchBox';
import MovieBox from './MovieBox';
import HeaderFooter from './HeaderFooter';

import common from './../../styles/common.css';
import styles from './../../styles/header.css';

class Header extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    const movieOpend = this.props.movieOpend;
    const emptyResults = this.props.emptyResults;

    let headerBody = null;
    if (movieOpend) {
      headerBody = <MovieBox />;
    } else {
      headerBody = <SearchBox />;
    }

    return (
      <header className = {styles.root}>
        <div className = {common.wrapper}>
          <Nav movieOpend={movieOpend}/>
          {headerBody}
        </div>
        <HeaderFooter movieOpend={movieOpend} emptyResults = {emptyResults}/>
      </header>
    );
  }

}

export default Header;
