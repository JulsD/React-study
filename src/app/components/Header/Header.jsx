import React from 'react';

import { HeaderNav, SearchBox, MovieBox, HeaderFooter } from './../';
import common from './../../../styles/common.css';
import styles from './header.css';

class Header extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    const movieOpend = this.props.movieOpend;
    const emptyResults = this.props.emptyResults;

    let headerBody = null;
    if (movieOpend) {
      headerBody = <MovieBox />;
    } else {
      headerBody = <SearchBox />;
    }

    return (
      <header className = {styles.root}>
        <div className = {common.wrapper}>
          <HeaderNav movieOpend={movieOpend}/>
          {headerBody}
        </div>
        <HeaderFooter movieOpend={movieOpend} emptyResults = {emptyResults}/>
      </header>
    );
  }

}

export default Header;
