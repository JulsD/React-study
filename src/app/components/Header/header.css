.root {
  position: relative;
  width: 100%;
  color: #fff;
  background-color: rgba(0,0,0,.7);
}
.root:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-image: url(https://assets.nflxext.com/ffe/siteui/vlv3/dc18d1c3-9e06-41a4-a5b0-409388e3f9e8/df3756f5-3cf4-44b5-b51c-cdfcac3810dc/PL-en-20170911-popsignuptwoweeks-perspective_alpha_website_small.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  z-index: -1;
}
