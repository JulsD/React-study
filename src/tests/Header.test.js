import React from 'react';
import Header from '../app/components/Header';

test('Header should render a <header>', () => {
  const wrapper = shallow(
    <Header />
  );
  expect(wrapper).toMatchSnapshot();
});
