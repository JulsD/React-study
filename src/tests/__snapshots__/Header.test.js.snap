// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a <header> 1`] = `
<header
  className="root"
>
  <Wrapper />
</header>
`;
